File: app/src/main/java/com/jameschamberlain/footballteamtracker/team/TeamRecyclerAdapter.java
package com.jameschamberlain.footballteamtracker.team;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.jameschamberlain.footballteamtracker.Player;
import com.jameschamberlain.footballteamtracker.R;

import java.util.ArrayList;
import java.util.Locale;

public class TeamRecyclerAdapter extends RecyclerView.Adapter<TeamRecyclerAdapter.ViewHolder> {


    private ArrayList<Player> players;
    private Fragment parentFragment;

    TeamRecyclerAdapter(ArrayList<Player> players, Fragment fragment) {
        this.players = players;
        this.parentFragment = fragment;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_player_details, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Player currentPlayer = players.get(position);
        holder.name.setText(currentPlayer.getName());
        holder.goals.setText(String.format(Locale.ENGLISH, "Goals: %d", currentPlayer.getGoals()));
        holder.assists.setText(String.format(Locale.ENGLISH, "Assists: %d", currentPlayer.getAssists()));
        holder.parentLayout.setOnClickListener(new playerOnClickListener(parentFragment, currentPlayer));
    }

    @Override
    public int getItemCount() {
        return players.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView name;
        TextView goals;
        TextView assists;
        RelativeLayout parentLayout;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name_text_view);
            goals = itemView.findViewById(R.id.goals_text_view);
            assists = itemView.findViewById(R.id.assists_text_view);
            parentLayout = itemView.findViewById(R.id.parent_layout);
        }
    }


}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/team/TeamRecyclerAdapter.kt
package com.jameschamberlain.footballteamtracker.team

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.jameschamberlain.footballteamtracker.objects.Player
import com.jameschamberlain.footballteamtracker.databinding.ItemPlayerDetailsBinding
import java.util.*

class TeamRecyclerAdapter internal constructor(private val players: ArrayList<Player>, private val parentFragment: Fragment) : RecyclerView.Adapter<TeamRecyclerAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemBinding = ItemPlayerDetailsBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentPlayer = players[position]
        holder.name.text = currentPlayer.name
        holder.goals.text = String.format(Locale.ENGLISH, "Goals: %d", currentPlayer.goals)
        holder.assists.text = String.format(Locale.ENGLISH, "Assists: %d", currentPlayer.assists)
//        holder.parentLayout.setOnClickListener(PlayerOnClickListener(parentFragment, currentPlayer))
    }

    override fun getItemCount(): Int {
        return players.size
    }

    inner class ViewHolder(itemBinding: ItemPlayerDetailsBinding) : RecyclerView.ViewHolder(itemBinding.root) {
        var name: TextView = itemBinding.nameTextView
        var goals: TextView = itemBinding.goalsTextView
        var assists: TextView = itemBinding.assistsTextView
        var parentLayout: ConstraintLayout = itemBinding.root

    }

}